packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -206,7 +210,14 @@ class FileCollection extends DocumentCollection {
       throw new Error('missing name argument')
     }
 
-    const path = uri`/files/${fileId}?Name=${name}&Type=file&Executable=${executable}`
+    let metadataId = ''
+    let path = uri`/files/${fileId}?Name=${name}&Type=file&Executable=${executable}`
+    if (metadata) {
+      const meta = await this.createFileMetadata(metadata)
+      metadataId = meta.data.id
+      path = path + `&MetadataID=${metadataId}`
```

Comment:
It surprises me that we use a path component to add the metadata id to the io.cozy.files instead of using POST data. What is the reason behind this ?

Reply:
We already use the body to send the content of the file. So we have 2 choices : 
- Use an object in a query string to send the metadata 
- create a new header http 

In both case, we have a size limitation depending of the browser. So we had this current workaround where we create the metadata before (no size limit since we can use the body for that) and after we use the ID.

Reply:
I think a comment with a part of this explanation would be beneficial.